lib/k8s/stack.rb

```diff
@@ -44,37 +47,33 @@ def self.delete(name, client, **options)
 
     attr_reader :name, :resources
 
-    def initialize(name, resources = [], debug: false, label: self.class::LABEL, checksum_annotation: self.class::CHECKSUM_ANNOTATION)
+    def initialize(name, resources = [], debug: false, label: self.class::LABEL, checksum_annotation: self.class::CHECKSUM_ANNOTATION, last_configuration_annotation: self.class::LAST_CONFIG_ANNOTATION)
       @name = name
       @resources = resources
       @keep_resources = {}
       @label = label
       @checksum_annotation = checksum_annotation
+      @last_config_annotation = last_configuration_annotation
 
       logger! progname: name, debug: debug
     end
 
-    # Random "checksum" used to identify different stack resource versions using an annotation.
-    #
-    # NOTE: This is not actually a checksum.
-    #
-    # @return [String]
-    def checksum
-      @checksum ||= SecureRandom.hex(16)
-    end
-
     # @param resource [K8s::Resource] to apply
     # @param base_resource [K8s::Resource] preserve existing attributes from base resource
     # @return [K8s::Resource]
     def prepare_resource(resource, base_resource: nil)
-      if base_resource
-        resource = base_resource.merge(resource)
-      end
+      # XXX: base_resource is not really used anymore, kept for backwards compatibility for a while
+
+      # calculate checksum  only from the "local" source
+      checksum = resource.checksum
 
       # add stack metadata
       resource.merge(metadata: {
         labels: { @label => name },
-        annotations: { @checksum_annotation => checksum },
+        annotations: {
+          @checksum_annotation => checksum,
+          @last_config_annotation => resource.to_json
```

Comment:
Are we using this somewhere?

Reply:
Nevermind, yes we are 😄

Reply:
Indeed, we are using it. Although it was used on `K8s::Resource` as a hardcoded string, now fixed in the later commits.